algorithms/kernel/svm/svm_train_thunder_cache.h

```diff
@@ -0,0 +1,217 @@
+/* file: svm_train_thunder_cache.h */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  SVM cache structure implementation
+//--
+*/
+
+#ifndef __SVM_TRAIN_THUNDER_CACHE_H__
+#define __SVM_TRAIN_THUNDER_CACHE_H__
+
+#include "service/kernel/service_utils.h"
+#include "externals/service_memory.h"
+#include "service/kernel/data_management/service_micro_table.h"
+#include "service/kernel/data_management/service_numeric_table.h"
+#include "data_management/data/numeric_table_sycl_homogen.h"
+#include "algorithms/kernel/svm/svm_train_cache.h"
+#include "externals/service_service.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace svm
+{
+namespace training
+{
+namespace internal
+{
+using namespace daal::data_management;
+
+/**
+ * Common interface for cache that stores kernel function values
+ */
+template <typename algorithmFPType, CpuType cpu>
+class SVMCacheIface<thunder, algorithmFPType, cpu> : public SVMCacheCommonIface<algorithmFPType, cpu>
```

Comment:
Class name confuses a little bit, because class contains data fields while have Iface in the name.